src/views/LoginSuccess.vue

```diff
@@ -66,21 +66,37 @@ export default class LoginSuccess extends Vue {
 
     private result?: LoginResult;
 
+    private render: boolean = false;
+    private retrievalComplete: boolean = false;
+
     private async mounted() {
+        const currentWalletInfo = await WalletStore.Instance.get(this.keyguardResult.keyId);
+
+        // if there is a WalletInfo for the id already, use it
+        if (currentWalletInfo && currentWalletInfo.label) {
+            this.walletLabel = currentWalletInfo.label;
+            if (currentWalletInfo.accounts) {
+                this.accounts = currentWalletInfo.accounts;
+            }
+        }
+
         // The Keyguard always returns (at least) one derived Address,
         // thus we can already create a complete WalletInfo object that
         // can be displayed while waiting for the network.
+        // before adding, make sure it is not set already
         this.keyguardResult.addresses.forEach((addressObj) => {
             const address = new Nimiq.Address(addressObj.address);
-            const addressInfo = new AccountInfo(
-                addressObj.keyPath,
-                this.defaultAccountLabel,
-                address,
-            );
-            this.accounts.set(addressInfo.userFriendlyAddress, addressInfo);
-            this.accountsUpdateCount += 1; // Trigger DOM update via computed property `this.accountsArray`
+            if (!this.accounts.has(address.toUserFriendlyAddress())) {
+                const addressInfo = new AccountInfo(
+                    addressObj.keyPath,
+                    this.defaultAccountLabel,
+                    address,
+                );
+                this.accounts.set(addressInfo.userFriendlyAddress, addressInfo);
+            }
         });
-
+        this.accountsUpdateCount += 1; // Trigger DOM update via computed property `this.accountsArray`
+        this.render = true; // trigger rendering now, when variables are set up correctly.
```

Comment:
I would prefer an adjective instead of 'render', like 'contentReady' or whatever.